fuse_models/include/fuse_models/odometry_2d_publisher.h

```diff
@@ -204,6 +205,16 @@ class Odometry2DPublisher : public fuse_core::AsyncPublisher
                                                                       //!< throttle messages, that can be reset on start
 
   ros::Timer publish_timer_;
+
+  ros::NodeHandle publish_timer_node_handle_;        //!< A dedicated node handle for the publish timer, so it can use
+                                                     //!< its own callback queue
+  ros::AsyncSpinner publish_timer_spinner_;          //!< A dedicated async spinner for the publish timer that manages
+                                                     //!< its callback queue with a dedicated thread
+  ros::CallbackQueue publish_timer_callback_queue_;  //!< A dedicated callback queue for the publish timer
```

Comment:
Since `publish_timer_callback_queue_` is passed as an argument to `publish_timer_spinner_`'s constructor, it needs to appear in the header file first.

And since `publish_timer_node_handle_` is configured to use `publish_timer_callback_queue_`, I'm curious if `publish_timer_callback_queue_` should really come before `publish_timer_node_handle_` as well.

```suggestion
  ros::CallbackQueue publish_timer_callback_queue_;  //!< A dedicated callback queue for the publish timer
  ros::NodeHandle publish_timer_node_handle_;        //!< A dedicated node handle for the publish timer, so it can use
                                                     //!< its own callback queue
  ros::AsyncSpinner publish_timer_spinner_;          //!< A dedicated async spinner for the publish timer that manages
                                                     //!< its callback queue with a dedicated thread
```

Reply:
Makes sense. Indeed, your suggestion matches the order in https://github.com/locusrobotics/fuse/blob/2239cffa1941861588856117388fea59b87f6c1e/fuse_core/include/fuse_core/async_publisher.h#L136-L140